File: backend/models/Message.js
import mongoose from 'mongoose';

const messageSchema = new mongoose.Schema({
  sender: {
    type: String,
    enum: ['USER', 'BOT'],
    required: true,
  },
  message: {
    type: String,
    required: true,
  },
  messageType: {
    type: String,
    default: 'text',
  },
  timestamp: {
    type: Date,
    default: Date.now,
  },
});

export default mongoose.model('Message', messageSchema);

File: chatbot-backend/db/message.py
# models/message.py
from pydantic import BaseModel
from datetime import datetime
from typing import Literal

class Message(BaseModel):
    sender: Literal["USER", "BOT"]
    message: str
    messageType: str = "text"
    timestamp: datetime = datetime.now()

    class Config:
        json_schema_extra = {
            "example": {
                "sender": "USER",
                "message": "Xin chào!",
                "messageType": "text",
            }
        }